util/text.jsx

```diff
@@ -38,20 +55,33 @@ export function insertSpans(text, highlights, options = { p: true }) {
         );
 
       const regexStr = highlight.regex || escapeRegex(highlight.text);
-      const regex = new RegExp(regexStr, 'igd');
+      let regex;
+      try {
+        regex = new RegExp(regexStr, 'igd');
+      } catch {
+        // error handling in case of old safari, which doesn't accept `d` flag for regex
+        regex = new RegExp(regexStr, 'ig');
+      }
 
       let match;
       // eslint-disable-next-line no-cond-assign
       while ((match = regex.exec(text))) {
+        const regexText = match[0];
+
         // Add all capture groups (but not the whole string) to the list
-        const { indices } = match;
+        const { indices = null, index } = match;
+
         arr.push(
-          ...match.slice(1).map((group, i) => ({
-            ...highlight,
-            match: group,
-            start: indices[i + 1][0],
-            end: indices[i + 1][1],
-          }))
+          ...match.slice(1).map((group, i) => {
+            const { start, end } = getIndices({ i, regexText, group, indices, index });
```

Comment:
Imo, this would be *slightly* more readable if you moved the conditional out of the function like this:
```suggestion
            const [ start, end ] = indices ?? getIndices({ i, regexText, group, index });
```

Reply:
Ooh! Nullish coalescing operator in the wild!

Agreed with @pandringa.